src/access/thread-open-mod-access.js

```diff
@@ -0,0 +1,133 @@
+'use strict'
+
+const AccessController = require('orbit-db-access-controllers/src/access-controller-interface')
+const ensureAddress = require('orbit-db-access-controllers/src/utils/ensure-ac-address')
+const entryIPFS = require('ipfs-log/src/entry')
+
+const type = 'thread-access'
+
+// TODO need extend access controller interface?
+class ThreadAccessController {
+  constructor (orbitdb, ipfs, options) {
+    // super()
+    this._orbitdb = orbitdb
+    this._db = null
+    this._options = options || {}
+    this._ipfs = ipfs
+    this._members = options.members
+    this._rootMod = options.rootMod
+    this._threadName = options.threadName
+  }
+
+  // Returns the type of the access controller
+  static get type () { return type }
+
+  // Returns the address of the OrbitDB used as the AC
+  get address () {
+    return this._db.address
+  }
+
+  async canAppend (entry, identityProvider) {
+    const trueIfValidSig = async () => await identityProvider.verifyIdentity(entry.identity)
+
+    const op = entry.payload.op
+    const mods = this.capabilities['mod']
+    const members = this.capabilities['member']
+    const isMod = members.includes(entry.identity.id)
+    const isMember = members.includes(entry.identity.id)
+
+    if (op === 'ADD') {
+      // Anyone can add entry if open thread
+      if (!this._members) return await trueIfValidSig()
+      // Not open thread, any member or mod can add to thread
+      if (isMember || isMod) return await trueIfValidSig()
+    }
+
+    if (op === 'DEL') {
+      const hash = entry.payload.value
+      const delEntry = await entryIPFS.fromMultihash(this._ipfs, hash)
+
+      // An id can delete their own entries
+      if (delEntry.identity.id === entry.identity.id) return await trueIfValidSig()
+
+      // Mods can't delete other mods entries
+      if (mods.includes(delEntry.identity.id)) return false
+
+      // Mods can delete any other entries
+      if (isMod) return await trueIfValidSig()
+    }
+
+    return false
+  }
+
+  get capabilities () {
+    // TODO dont do this for every entry, save result, update on db change
```

Comment:
:+1:

Reply:
not only updated at first load or on AC updates, instead of each entry and for each members/mods list